### PR TITLE
Add AllReady, a real-world ASP.NET Core app to our suite

### PR DESCRIPTION
**Pull request by adamsitnik**
Hello everyone!

This PR introduces a new app to our benchmarking suite: AllReady.

### About AllReady

"allReady is an open-source solution focused on increasing awareness, efficiency and impact of preparedness campaigns as they are delivered by humanitarian and disaster response organizations in local communities. http://www.htbox.org/projects/allready"

### Why AllReady?

It's a real-world web app which uses some of the most popular libraries like `MediatR`, `Autofac`, `Hangfire`, `EntityFrameworkCore`, `Newtonsoft.Json` and `WindowsAzure.Storage`. It should help us to be closer to what our end customers use in their production systems.

Our goal is to have a real-world web application scenario to optimize and test the performance for real workloads.

### Benchmark

The benchmark does exactly the same job the MusicStore benchmark does. I have even refactored the code and now it's shared between these two apps.

### How to review this big PR.

99% of the code is just a copied code from the AllReady repostiory.

I suggest that the reviewers review separate commits, not all the code in the PR.

So the commits are:

1. [initial copy of AllReady](https://github.com/aspnet/JitBench/commit/0b02fd2fa4e005733818900848f6bbc4906e6f64) - here I have simply copy-pasted all the code from AllReady repository. No manual changes
2. [merge all 2 types from AllReady.Core.csproj to the web app project](https://github.com/aspnet/JitBench/commit/716927967aaeb88d1b514db40fc3d9b831d3c983) - originally `AllReady.csproj` was referencing `AllReady.Core.csproj` project. Since the project contained only two types, I decided to "merge" them into main project. Just to keep it simple.
3. [add AllReady to JitBench.sln](https://github.com/aspnet/JitBench/commit/d014810bec6c5cd81c94f69898a2dcb3128707c3) - I just added the new `.csproj` to `JitBench.sln`. **Question:** is it ok that I have also updated the VS version?
4. [keep all javascript dependencies in the repo, remove npm & gulp steps to make it independent from js changes and 3rd party tools like npm](https://github.com/aspnet/JitBench/commit/59d62843d271f365d53843dd2fb1dcb4a5e27f1b) AllReady is a modern web app and it uses some fancy js tools like NPM (the js package manager) and gulp. We want to run the benchmarks on our CI machines and my guess what that we don't want to install any js tools on our machines. Moreover, the `npm install` which was part of the `prepublish` step needed at least few minutes to download all js dependencies. **What I did was: download all the js dependencies, build the js part on my machine and make it part of our repo**. The advantage is that now we don't need anything else than `dotnet cli` to run this app. The disadvantage is that I added a lot of file to the GIT repo.
5. [use InMemoryDb to avoid having a SQL Server running on CI as a requirement](https://github.com/aspnet/JitBench/commit/8935bfe67bc1146497082628c179f3b7d2c8d075) - I have used InMemory providers for EntityFramework and Hangfire.  To once again avoid dependencies and keep our benchmarks more stable accross the runs.
6. [move all common web app settings to webapp.props file](https://github.com/aspnet/JitBench/commit/3a442067aa04acd4bd39c15c1eddab106f2ce4d8) - I moved all ASP.NET settings to a single props file to avoid any duplicates and differences in the future.
7. [move common benchmarking code to separate class, reuse it](https://github.com/aspnet/JitBench/commit/8f0149bbb675ad1e5080b25f122a50ea30bdb020) - I have reused the benchmarking code from MusicStore. I decided to reference the common class as **link** to avoid project dependencies and keep our build simpler.
8. [update the script and README](https://github.com/aspnet/JitBench/commit/4037026074aa9f71c737042b7469db17007b0801) - I updated the README and script. The script by default runs MusicStore so it is not breaking change.

/cc @jorive @noahfalk

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,6 +102,12 @@ This runs steps 1-5 in the -WhatIf output from above. You can now run dotnet Mus
 
 ## Other things you can do
 
+### Use AllReady app
+
+MusicStore is the default app being executed with `RunBenchmark.ps1`. If you want to try the `AllReady` app you need to provide two extra arguments to the script:
+
+`-WebAppFolder src\AllReady -WebAppDllName AllReady.dll`
```
Comment (noahfalk) [inline README.md]: I suggest making these command lines as simple as possible. For example add just one arg that is focused on intent and make it the script's responsibility to know what source folder / dll to use.

-test [AllReady|MusicStore]

If not specified MusicStore remains the default.

**Comment by noahfalk** [comment]
@kouvel should probably be reviewer rather than me (I've stepped away from tiered jitting for a little while at least)
@rynowak - keeping you in the loop

**Comment by adamsitnik** [comment]
@noahfalk good point! I have simplified the script

**Comment by jorive** [comment]
If at all possible, we should avoid bringing the source code here. Instead, we could use submodule, download the zipped repo, etc.

**Comment by adamsitnik** [comment]
Closing this one since I was able to get it done with `git submodules` as suggested by @jorive #86 is the new PR

**Pull request closed without merging**